mycore-base/src/main/java/org/mycore/datamodel/language/MCRLanguageResolver.java

```diff
@@ -24,21 +24,30 @@
 import javax.xml.transform.TransformerException;
 import javax.xml.transform.URIResolver;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.jdom2.Document;
 import org.jdom2.Element;
 import org.jdom2.transform.JDOMSource;
 import org.mycore.tools.MCRLanguageOrientationHelper;
 
 /**
  * Resolves languages by code. Syntax: language:{ISOCode}
- * 
+ *
  * @author Frank Lützenkirchen
  */
 public class MCRLanguageResolver implements URIResolver {
 
+    private static final Logger LOGGER = LogManager.getLogger();
+
     public Source resolve(String href, String base) throws TransformerException {
         try {
-            String code = href.split(":")[1];
+            String[] hrefContent = href.split(":");
+            if (hrefContent.length < 2) {
```

Comment:
I think it would be better if this method throws an error if the parameter is missing. The error should be IllegalArgumentException. The iso code should not be optional. If you want to ignore the exception, than you should use notNull:language:... , which should catch the exception.